### DISCUSSION TITLE
Automatically upload last backup tar to the server

### DISCUSSION THREAD

**Discussion by Hariko50** [GetPublii/Publii#1909] [Help] [answered]
I would like to automatically upload the last backup.tar of the website when uploading the files. Better safe than sorry. Does anyone have an idea where I should start?

**Comment by Hariko50** [comment]
Does Publii have keyboard shortcuts to start the backup, for example?

**Reply by michaelfluegge** [reply]
I save the backup files into a folder that is synchronized with Resilio sync to a NAS automatically. There are several other tools like SyncThing doing such jobs. Or you go for something like Backblaze, which safes your drive to the web. Better safe than sorry:-)

**Reply by Hariko50** [reply]
That's a practical option, then the only thing missing is a way for publii to automatically create backup files. I rarely think about doing this manually.

**Reply by michaelfluegge** [reply]
You will do better after the first crash. Until then I would go for safe soon, safe often.

**Reply by Hariko50** [reply]
The function is already built in, it would only be necessary to add a code that automatically creates a backup when the APP is opened or closed.

**Reply by michaelfluegge** [reply]
Ok, I did not know that. But if so, I would like to have some functionality come with that. I have several sites online, and I don’t want each of them backed up every time I close Publii. Only the one I worked on.

**Reply by bttrx** [reply, marked as answer]
@Hariko50 Publii is no backup software. As a workaround you could create a script for starting Publii and include a command similar to this. (I'm assuming Linux here.)

```bash
tar -cf SITENAME-$(date +%Y-%m-%d-%H-%M-%S).tar -C /home/USERNAME/Dokumente/Publii/sites/SITENAME input
```

(You have to modify SITENAME & USERNAME to match your system. 'Dokumente' maybe too.)
This will create a timestamped .tar file to the current folder.

**Comment by Hariko50** [comment]
Your suggestion has inspired me to switch from Windows back to Linux.

I’ve created a Bash script to automate the backup of a specific folder (input) from my Publii site and upload it to a remote server. The script monitors another folder (output) for changes and only performs the backup if modifications are detected. Here’s a breakdown of what it does:

What the Script Does:
Monitors the output Folder:

The script continuously monitors the /home/USERNAME/Dokumente/Publii/sites/SITENAME/output folder for changes.

It checks the folder’s modification timestamp every 60 seconds (configurable).

Creates a Backup of the input Folder:

If changes in the output folder are detected, the script creates a .tar archive of the input folder.

The backup file is named with the site name and the current timestamp (e.g., SITENAME-2023-10-05-15-27-54.tar).

Saves the Backup Locally:

The .tar file is saved in the /home/USERNAME/Dokumente/Publii/backups directory.

Uploads the Backup to a Remote Server:

The script uses scp to upload the .tar file to a remote server.

The backup is saved in the /REMOTE_DIR/backup directory on the remote server.

Runs Continuously:

The script runs in an infinite loop, monitoring the output folder and performing backups of the input folder as needed.

To Run the Script in the Background:
To ensure the script continues running even after closing the terminal, use the nohup command and I start it automatically with cron
